src/axolotl/utils/config/models/input/v0_4_1/__init__.py

```
@@ -1402,6 +1403,43 @@ def check_torch_compile_deepspeed(cls, data):
             )
         return data
 
+    @model_validator(mode="before")
+    @classmethod
+    def check_npu_config(cls, data):
+        if is_torch_npu_available():
+            # check attention config
+            attn_list = ["flash_attention", "sdp_attention", "s2_attention"]
+            for attn in attn_list:
+                if data.get(attn):
+                    data[attn] = False
+                    LOG.warning(
+                        f"{attn} is currently not supported in Ascend npu, disabling this configuration."
+                    )
+
+            # check quant config
+            if data.get("optimizer") is not None and "bit" in data.get("optimizer"):
+                optimizer = data.get("optimizer")
+                raise NotImplementedError(
+                    f"{optimizer} is currently not supported in Ascend npu, choose another one please."
+                )
+
+            quant_list = ["load_in_8bit", "load_in_4bit"]
+            for quant in quant_list:
+                if data.get(quant):
+                    data[quant] = False
+                    LOG.warning(
+                        "Quantification is currently not supported in Ascend npu, disabling this configuration."
+                    )
+
+            # check dtype config
+            if data.get("tf32"):
+                data["tf32"] = False
+                LOG.warning(
+                    "tf32 dtype is currently not supported in Ascend npu, disabling this configuration"
+                )
```

Should this just error out instead of warn?